lib/libc/newlib/libc-hooks.c

```diff
@@ -138,6 +138,12 @@ SYS_INIT(malloc_prepare, POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT);
 /* Current offset from HEAP_BASE of unused memory */
 LIBC_BSS static size_t heap_sz;
 
+void get_heap_statistics(size_t *total, size_t *used)
+{
+	*total = MAX_HEAP_SIZE;
+	*used = heap_sz;
+}
```

Comment:
This is not really an "API." At most, it is a random function declared in the Newlib hook source file.

In order to be qualified as an "API," this function should be:

* defined in one of the public headers under `include/zephyr/`.
* prefixed (e.g. `libc_`).
* properly documented (with Doxygen comments).

Note that the libc integration layer does not currently have an "API" -- we have `libc-hooks.h`, which is a private header for accessing the libc integration layer internals; but, that is not really a public API header meant to be directly used by applications.

We _could_ introduce a public API header (e.g. `include/zephyr/libc.h`) for the libc integration layer to provide functions like these; but, its purpose and scope needs to be further investigated and discussed before we should do that.

Reply:
@qipengzha
Note that if you know the total size, the current arena size (equivalent to `used`) can be retrieved with libc's `mallinfo().arena`.  `mallinfo()` also returns other useful statistics like `.fordblks` that informs you how much of the space allocated from the system is actually free blocks.

Reply:
`mallinfo()` is great - I kind of wish we had that in Zephyr too. However, since we're moving to a common heap implementation in Zephyr, that might be a distinct possibility in the near future.

Reply:
We could certainly add some kind of informational function about malloc, but `mallinfo2` contains a lot of very glibc-specific details about the malloc arena. Newlib and picolibc both provide this function and fake out the unused fields, maybe that would be good enough?

Reply:
thanks all your good suggestions,  I will check to use this standard libc api.